### PR TITLE
Rebase against Element Android v1.5.2

### PR DESCRIPTION
**Pull request by appndigital**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : rebase

## Content

Fix conflicts from rebase 1.4.36

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 13

## Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [x] Accessibility has been taken into account. See https://github.com/tchapgouv/tchap-android-v2/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [TCHAP_CHANGES.md](https://github.com/tchapgouv/tchap-android-v2/blob/develop/TCHAP_CHANGES.md)
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/tchapgouv/tchap-android-v2/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/tchapgouv/tchap-android-v2/blob/develop/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/HomeActivity.kt [hunk 1]
```diff
@@ -593,14 +585,8 @@ class HomeActivity :
                     .setPositiveButton(R.string.yes) { _, _ -> bugReporter.openBugReportScreen(this) }
                     .setNegativeButton(R.string.no) { _, _ -> bugReporter.deleteCrashFile() }
                     .show()
-<<<<<<< HEAD
-        } else if (shouldShowDisclaimerDialog(this)) {
-            showDisclaimerDialog(this)
-            homeActivityViewModel.handle(HomeActivityViewActions.DisclaimerDialogShown)
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeActivity.kt:599]: we should keep this condition which was specific to tchap

File: vector/src/main/java/im/vector/app/features/disclaimer/DisclaimerDialog.kt [hunk 2]
```diff
@@ -21,48 +21,16 @@ import android.content.SharedPreferences
...
-    }
-
-    val dialogLayout = activity.layoutInflater.inflate(R.layout.dialog_disclaimer_content, null)
-
-    MaterialAlertDialogBuilder(activity)
-            .setView(dialogLayout)
-            .setCancelable(false)
-            .setNeutralButton(R.string.ok, null)
-            .show()
-}
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/disclaimer/DisclaimerDialog.kt:58]: please restore this code

File: vector/build.gradle [hunk 3]
```diff
@@ -465,8 +451,6 @@ dependencies {
         exclude group: "org.jetbrains.kotlinx", module: "kotlinx-coroutines-debug"
     }
 
-<<<<<<< HEAD
-
```
Comment (Florian14) [inline vector/build.gradle:469]: All the following lines should probably be moved to `vector-app/build.gradle` if this is the case in Element

File: vector/build.gradle [hunk 4]
```diff
@@ -208,21 +205,12 @@ android {
...
-//        }
-=======
->>>>>>> v1.5.2
     }
 
     buildFeatures {
         viewBinding true
     }
 }
-<<<<<<< HEAD
```
Comment (Florian14) [inline vector/build.gradle:225]: same as my next comment (maybe move it to `vector-app/build.gradle`)

File: vector/build.gradle [hunk 5]
```diff
@@ -85,7 +85,6 @@ android {
                 testCoverageEnabled = coverage.enableTestCoverage
             }
         }
-<<<<<<< HEAD
         // Tchap: No nightly
 //        nightly {
```
Comment (Florian14) [inline vector/build.gradle:89]: We can probably remove the nightly bloc if this is not present anymore in this gradle file in Element

File: vector-app/build.gradle [hunk 6]
```diff
@@ -309,21 +301,11 @@ android {
...
     // The 'target' dimension permits to specify which platform are used
     // The 'voip' flavor dimension permits to include/exclude jitsi at compilation time
     // The 'pinning' flavor dimension permits to enable/disable certificate pinning with fingerprint check
     flavorDimensions "store", "target", "voip", "pinning"
-=======
-    sourceSets {
-        nightly {
-            java.srcDirs += "src/release/java"
-        }
-    }
```
Comment (Florian14) [inline vector-app/build.gradle:323]: please comment these lines and move them upon the flavor dimensions, as it is in Element

File: vector-app/build.gradle [hunk 7]
```diff
@@ -31,20 +31,12 @@ knit {
...
 ext.versionMajor = 2
-=======
-ext.versionMajor = 1
->>>>>>> v1.5.2
 ext.versionMinor = 5
 // Note: even values are reserved for regular release, odd values for hotfix release.
 // When creating a hotfix, you should decrease the value, since the current value
 // is the value for the next regular release.
-<<<<<<< HEAD
 ext.versionPatch = 0
```
Comment (Florian14) [inline vector-app/build.gradle:39]: you forgot to increment this digit after the 2.5.0, you should consider it after the rebase.

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt [hunk 8]
```diff
@@ -114,13 +114,9 @@ class CommonTestHelper internal constructor(context: Context, val cryptoConfig:
                     MatrixConfiguration(
                             applicationFlavor = "TestFlavor",
                             roomDisplayNameFallbackProvider = TestRoomDisplayNameFallbackProvider(),
-<<<<<<< HEAD
+                            syncConfig = org.matrix.android.sdk.api.SyncConfig(longPollTimeout = 5_000L),
```
Comment (Florian14) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt:117]: why don't we have the import?

File: library/ui-strings/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1242,16 +1233,11 @@
...
     <string name="encryption_import_import">Import</string>
-<<<<<<< HEAD
     <string name="encryption_never_send_to_unverified_devices_title">Encrypt to verified devices only</string> <!-- Tchap: Use custom string -->
     <string name="encryption_never_send_to_unverified_devices_summary">Never send encrypted messages to unverified devices from this device.</string> <!-- Tchap: Use custom string -->
-=======
-    <string name="encryption_never_send_to_unverified_devices_title">Encrypt to verified sessions only</string>
-    <string name="encryption_never_send_to_unverified_devices_summary">Never send encrypted messages to unverified sessions from this session.</string>
     <string name="encryption_never_send_to_unverified_devices_in_room">Never send encrypted messages to unverified sessions in this room.</string>
     <string name="some_devices_will_not_be_able_to_decrypt">⚠ There are unverified devices in this room, they won’t be able to decrypt messages you send.</string>
     <string name="room_settings_global_block_unverified_info_text">🔒 You have enabled encrypt to verified sessions only for all rooms in Security Settings.</string>
```
Comment (Florian14) [inline library/ui-strings/src/main/res/values/strings.xml:1240]: In Tchap, we replace `session` with `device`, please can you update these new strings (we should consider in the future if we want to switch back to `session`)

File: coverage.gradle [hunk 10]
```diff
@@ -80,21 +80,12 @@ task generateCoverageReport(type: JacocoReport) {
 
 task unitTestsWithCoverage(type: GradleBuild) {
     // the 7.1.3 android gradle plugin has a bug where enableTestCoverage generates invalid coverage
-<<<<<<< HEAD
     startParameter.projectProperties.coverage = [enableTestCoverage:  false]
     tasks = [':vector:testGplayBtchapWithoutvoipWithoutpinningDebugUnitTest', ':matrix-sdk-android:testDebugUnitTest']
```
Comment (Florian14) [inline coverage.gradle:84]: In the diff, Element only have `tasks = ['testDebugUnitTest']`, can we have the same? I would expect having a single task without module prefix here

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -158,30 +155,14 @@ You should consider adding Unit tests with your PR, and also integration tests (
...
->>>>>>> v1.5.2
 
 ##### Plurals
 
 Please use `plurals` resources when appropriate, and note that some languages have specific rules for `plurals`, so even if the string will always be at the plural form for English, please always create a `plurals` resource.
 
 Specific plural forms can be found [here](https://unicode-org.github.io/cldr-staging/charts/37/supplemental/language_plural_rules.html).
 
-<<<<<<< HEAD
-=======
```
Comment (Florian14) [inline CONTRIBUTING.md:184]: please keep it empty, the strings management is not relevant for Tchap

File: vector/src/main/java/im/vector/app/features/home/HomeActivity.kt (hunk 1 above)
Comment (appndigital) [reply vector/src/main/java/im/vector/app/features/home/HomeActivity.kt:599]: _shouldShowDisclaimerDialog_ and _showDisclaimerDialog_ are not available any more
Should I get back the code from older version ?

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt (hunk 8 above)
Comment (appndigital) [reply matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt:117]: What do you call import ?

File: library/ui-strings/src/main/res/values/strings.xml (hunk 9 above)
Comment (appndigital) [reply library/ui-strings/src/main/res/values/strings.xml:1240]: I check in the current tchap version : sometimes it uses session, sometimes device.
How do we know which term to use ?
How can I identify new strings ?

**Comment by giomfo** [comment]
related to #764

File: vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt [hunk 12]
```diff
@@ -68,11 +68,7 @@ class VectorActivityLifecycleCallbacks constructor(private val popupAlertManager
 
             // Get all activities from PermissionController module
             // See https://source.android.com/docs/core/architecture/modular-system/permissioncontroller#package-format
-<<<<<<< HEAD
             if (Build.VERSION.SDK_INT > Build.VERSION_CODES.S) {
-=======
-            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.S_V2) {
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt:74]: I think we can take this one, I don't see a reason to have a difference here

**Review by Florian14 — COMMENTED** [review]
Just to small remarks and it should be fine then

File: vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt (hunk 12 above)
Comment (appndigital) [reply vector/src/main/java/im/vector/app/features/lifecycle/VectorActivityLifecycleCallbacks.kt:74]: Build.VERSION_CODES.S_V2 doesn't exist in this project

**Review by Florian14 — APPROVED** [review]

**Pull request merged**